From eb1370ac6686799db76f5bc646a393d996e414e1 Mon Sep 17 00:00:00 2001
From: Hunseop Jeong <hs85.jeong@samsung.com>
Date: Thu, 21 Sep 2017 10:03:32 +0900
Subject: [PATCH] Fix ImageViewer to show the image correctly

ImageViewer didn't work because of image path. To fix the ImageViewer,
convert the image path correctly.

Signed-off-by: Hunseop Jeong <hs85.jeong@samsung.com>
---
 src/editor/ImageViewer.js       | 37 +++++++++++++++++++++++--------------
 src/htmlContent/image-view.html |  2 +-
 2 files changed, 24 insertions(+), 15 deletions(-)

diff --git a/src/editor/ImageViewer.js b/src/editor/ImageViewer.js
index 0b830ec..7fe91a9 100644
--- a/src/editor/ImageViewer.js
+++ b/src/editor/ImageViewer.js
@@ -29,6 +29,7 @@ define(function (require, exports, module) {
 
     var DocumentManager     = require("document/DocumentManager"),
         ImageViewTemplate   = require("text!htmlContent/image-view.html"),
+        PreferencesManager  = require("preferences/PreferencesManager"),
         ProjectManager      = require("project/ProjectManager"),
         LanguageManager     = require("language/LanguageManager"),
         MainViewFactory     = require("view/MainViewFactory"),
@@ -41,6 +42,13 @@ define(function (require, exports, module) {
 
     var _viewers = {};
 
+    function convertProjectPath(path) {
+        var newPath = path.split("/");
+        newPath.splice(2, 0, PreferencesManager.getViewState("projectId"));
+
+        return newPath.join("/");
+    }
+
     /**
      * ImageView objects are constructed when an image is opened
      * @see {@link Pane} for more information about where ImageViews are rendered
@@ -51,8 +59,7 @@ define(function (require, exports, module) {
      */
     function ImageView(file, $container) {
         this.file = file;
-        this.$el = $(Mustache.render(ImageViewTemplate, {fullPath: FileUtils.encodeFilePath(file.fullPath),
-                                                         now: new Date().valueOf()}));
+        this.$el = $(Mustache.render(ImageViewTemplate, {fullPath: convertProjectPath(file.fullPath), now: new Date().valueOf()}));
 
         $container.append(this.$el);
 
@@ -134,9 +141,7 @@ define(function (require, exports, module) {
                 }
                 var dimensionAndSize = dimensionString + sizeString;
                 self.$imageData.html(dimensionAndSize)
-                        .attr("title", dimensionAndSize
-                                    .replace("&times;", "x")
-                                    .replace("&mdash;", "-"));
+                    .attr("title", dimensionAndSize.replace("&times;", "x").replace("&mdash;", "-"));
             }
         });
 
@@ -147,7 +152,7 @@ define(function (require, exports, module) {
         this.$imageGuides.hide();
 
         this.$image.on("mousemove.ImageView", ".image-preview", _.bind(this._showImageTip, this))
-                   .on("mouseleave.ImageView", ".image-preview", _.bind(this._hideImageTip, this));
+            .on("mouseleave.ImageView", ".image-preview", _.bind(this._hideImageTip, this));
 
         this._updateScale();
     };
@@ -335,10 +340,12 @@ define(function (require, exports, module) {
             if (offsetX < (imagePos.left + imgWidth) &&
                     offsetY < (imagePos.top + imgHeight)) {
                 // Remember image scale div coordinates before hiding it.
-                this._scaleDivInfo = {left: scaleDivPos.left,
-                                 top: scaleDivPos.top,
-                                 right: scaleDivRight,
-                                 bottom: scaleDivBottom};
+                this._scaleDivInfo = {
+                    left: scaleDivPos.left,
+                    top: scaleDivPos.top,
+                    right: scaleDivRight,
+                    bottom: scaleDivBottom
+                };
                 this.$imageScale.hide();
             }
         }
@@ -372,10 +379,12 @@ define(function (require, exports, module) {
 
         // $view is "position:absolute" so
         //  we have to update the height, width and position
-        this.$el.css({top: pos.top + ((oHeight - iHeight) / 2),
-                        left: pos.left + ((oWidth - iWidth) / 2),
-                        width: iWidth,
-                        height: iHeight});
+        this.$el.css({
+            top: pos.top + ((oHeight - iHeight) / 2),
+            left: pos.left + ((oWidth - iWidth) / 2),
+            width: iWidth,
+            height: iHeight
+        });
         this._updateScale();
     };
 
diff --git a/src/htmlContent/image-view.html b/src/htmlContent/image-view.html
index 829a4cd..061640c 100644
--- a/src/htmlContent/image-view.html
+++ b/src/htmlContent/image-view.html
@@ -6,7 +6,7 @@
         </div>
         <div class="image">
             <div class="image-scale"></div>
-            <img class="image-preview" src="file:///{{fullPath}}?ver={{now}}">
+            <img class="image-preview" src="{{fullPath}}?ver={{now}}">
             <div class="image-tip">
                 <table class="tip-container">
                     <tr>
-- 
2.7.4

